From 0b03534010f5bcfcef371036bb2e63f090a7022d Mon Sep 17 00:00:00 2001
From: d34d <clark@cyngn.com>
Date: Mon, 13 Feb 2017 01:12:57 -0500
Subject: [PATCH 06/22] Show infinity for large notification counts

This patch adds a new string to SystemUI for displaying the infinity
symbol when a notification has a count larger than 999.  This only
affects the notification icon in the status bar and does not change
the text displayed in the actual notification.

These changes were manually cherry-picked from cm-13.0.

Change-Id: I3481aac13cd90ee20ed48194df3ec789c16c8b4b
---
 packages/SystemUI/res/values/cm_strings.xml                         | 6 ++++++
 .../src/com/android/systemui/statusbar/StatusBarIconView.java       | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/res/values/cm_strings.xml b/packages/SystemUI/res/values/cm_strings.xml
index 22fdd5d..b76d634 100644
--- a/packages/SystemUI/res/values/cm_strings.xml
+++ b/packages/SystemUI/res/values/cm_strings.xml
@@ -69,6 +69,12 @@
     <!-- Screen pinning dialog description (for devices without navbar) -->
     <string name="screen_pinning_description_no_navbar">This keeps it in view until you unpin. Touch &amp; hold Back to unpin.</string>
 
+    <!-- Text to use when the number in a notification info is too large
+         (greater than status_bar_notification_info_maxnum, defined in
+         values/config.xml) and must be truncated.
+         [CHAR LIMIT=4] -->
+    <string name="status_bar_notification_info_overflow" translatable="false">\u221E</string>
+
     <!-- Label for current user in power menu options dialog -->
     <string name="global_action_current_user">Current</string>
 
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
index 6cfd42f..907dec1 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
@@ -439,7 +439,7 @@ public class StatusBarIconView extends AnimatedImageView {
                 android.R.integer.status_bar_notification_info_maxnum);
         if (mIcon.number > tooBig) {
             str = getContext().getResources().getString(
-                        android.R.string.status_bar_notification_info_overflow);
+                        R.string.status_bar_notification_info_overflow);
         } else {
             NumberFormat f = NumberFormat.getIntegerInstance();
             str = f.format(mIcon.number);
-- 
2.7.4

